microbench/src/test/java/io/netty/microbench/util/AbstractExecutorMicrobenchmark.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.microbench.util;
+
+import static org.junit.Assert.assertNull;
+import io.netty.channel.EventLoop;
+import io.netty.util.concurrent.AbstractEventExecutor;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.ProgressivePromise;
+import io.netty.util.concurrent.Promise;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Fork;
+
+/**
+ * This harness facilitates the sharing of an executor between JMH and Netty and
+ * thus avoid measuring context switching in microbenchmarks.
+ */
+@Fork(AbstractExecutorMicrobenchmark.DEFAULT_FORKS)
+public class AbstractExecutorMicrobenchmark extends AbstractMicrobenchmarkBase {
+
+    protected static final int DEFAULT_FORKS = 0; // Forks has to be 0 so tasks are run immediately by JMH
+
+    public static void executor(EventLoop service) {
+        DelegateHarnessExecutor.executor(service);
```

Comment:
nifty! consider adding a comment here or in the class doc explaining how this magic is achieved